hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableAccountStore.java

```diff
@@ -177,10 +176,11 @@ private HederaAccount getAccountLeaf(@NonNull final AccountID id) {
                         if (alias.length() == EVM_ADDRESS_LEN && isMirror(alias)) {
                             yield fromMirror(alias);
                         } else {
-                            yield aliases.get(alias.asUtf8String()).num();
+                            final var entityNum = aliases.get(alias.asUtf8String());
+                            yield entityNum == null ? new EntityNumValue().num() : entityNum.num();
                         }
                     }
-                    case UNSET -> throw new RuntimeException("Account number not set in protobuf!!");
+                    case UNSET -> throw new InvalidOneOfKindException("Account number not set in protobuf!!");
```

Comment:
This isn't really an invalid kind, as we allow for UNSET.  I think this is more of a `MissingAccountNumberException`.

Reply:
It's OK as you change it for UnsetFieldException.  I tend to find more specific exceptions easier to deal with, but a generic exception works.